joomla/joomla-cms
# Add databasequeryerrors logging category

Based on dicussions in https://github.com/joomla/joomla-cms/pull/7183
#### Background

Currently, our database drivers are logging database query errors to the category `databasequery`.
However to actually see those appear in a log, you need to have several conditions met.
#### Rant

 I'm not even sure exactly as I got confused a lot while testing. I think you need to have global debug enabled and in the debug plugin also the "log queries" option enabled. But today it didn't work anymore for me. So I'm not sure how I got them to appear yesterday.
#### Solution

With this PR the errros would be logged as a new category "databasequeryerror", which will be logged as soon as you have the "Log Almost Everything" parameter enabled in the debug plugin.
With global debug on, you will get the query errors logged twice, because debug will try to run the query again in "explain" mode. That should be fine.

This PR also removes the logging which we introduced with some recently merged PRs. This is no longer neeeded.
It also adjusts the tooltip for the category parameter in the plugin. I thought it may be worth to add the category to the list there.
#### Testing
- Change a query so it fails. I just renamed the table name in the admin stats module (https://github.com/joomla/joomla-cms/blob/staging/administrator/modules/mod_stats_admin/helper.php#L84) to `#__user`. This module is loaded in the Control Panel by default if you have testing data installed so it was a simple test for me. But any query should work.
- Enable logging in the debug plugin. The most important is the "Log Almost Everything".
- Load the page and see if any errors are logged in the Joomla logfiles.
- Apply PR and load the page again. You now should see the database query error logged in the "logs/everything.php" logfile.
